plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergRegisterTableProcedure.java

```diff
@@ -0,0 +1,389 @@
+
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s')", TEST_SCHEMA_NAME, baseTableName),
+                ".*'TABLE_LOCATION' is missing.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', null)", TEST_SCHEMA_NAME),
+                ".*'TABLE_LOCATION' is missing.*");
+        assertQueryFails("CALL iceberg.system.register_table (null, null)",
+                ".*'TABLE_LOCATION' is missing.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s')", TEST_SCHEMA_NAME),
+                ".*'TABLE_NAME' is missing.*");
+        assertQueryFails("CALL iceberg.system.register_table (null)",
+                ".*'TABLE_NAME' is missing.*");
+        assertQueryFails("CALL iceberg.system.register_table ()",
+                ".*'SCHEMA_NAME' is missing.*");
+
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', null)", TEST_SCHEMA_NAME, baseTableName),
+                ".*table_location cannot be null.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', null, null)", TEST_SCHEMA_NAME),
+                ".*table_name cannot be null.*");
+        assertQueryFails("CALL iceberg.system.register_table (null, null, null)",
+                ".*schema_name cannot be null.*");
+
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", "", baseTableName, tableLocation),
+                ".*schemaName is empty.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", TEST_SCHEMA_NAME, "", tableLocation),
+                ".*tableName is empty.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", TEST_SCHEMA_NAME, baseTableName, ""),
+                ".*table_location is empty.*");
+        assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s', '%s')", TEST_SCHEMA_NAME, baseTableName, tableLocation, ""),
+                ".*metadata_file_name cannot be empty when provided as an argument.*");
+    }
+
+    @Test
+    public void testRegisterTableWithInvalidMetadataFileName()
+    {
+        String baseTableName = "test_register_table_with_invalid_metadata_file_name_" + randomTableSuffix();
+        String tableLocation = "/test/iceberg/hive";
+
+        String[] invalidMetadataFileNames = {
+                "/",
+                "../",
+                "../../",
+                "../../somefile.metadata.json",
+        };
+
+        for (String invalidMetadataFileName : invalidMetadataFileNames) {
+            assertQueryFails(format("CALL iceberg.system.register_table ('%s', '%s', '%s', '%s')", TEST_SCHEMA_NAME, baseTableName, tableLocation, invalidMetadataFileName),
+                    ".*is not a valid metadata file.*");
+        }
+    }
+
+    private static String trinoTableName(String tableName)
+    {
+        return format("%s.%s.%s", CATALOG, TEST_SCHEMA_NAME, tableName);
```

Comment:
We use this convention in product tests where table names need to be qualified.
here you don't need to qualify table names, since the test session has some current catalog and schema set.

Remove this method. Don't qualify names.

Reply:
Removed